metals/src/main/scala/scala/meta/internal/metals/MtagsResolver.scala

```diff
@@ -76,11 +88,79 @@ object MtagsResolver {
         )
         computed match {
           case State.Success(v) => Some(v)
-          case _: State.Failure => None
+          // Try to download latest supported snapshot
+          case _: State.Failure
+              if original.isEmpty && scalaVersion.contains("NIGHTLY") =>
+            findLatestSnapshot(scalaVersion) match {
+              case None => None
+              case Some(latestSnapshot) =>
+                scribe.warn(s"Using latest stable version $latestSnapshot")
+                resolve(
+                  latestSnapshot,
+                  Some(scalaVersion),
+                )
+            }
+          case _ =>
+            None
         }
       }
     }
 
+    /**
+     * Nightlies version are able to work with artifacts compiled within the
+     * same RC version.
+     *
+     * For example 3.2.2-RC1-bin-20221009-2052fc2-NIGHTLY presentation compiler
+     * will work with classfiles compiled with 3.2.2-RC1-bin-20220910-ac6cd1c-NIGHTLY
+     *
+     * @param exactVersion version we failed to find and looking for an alternative for
+     * @return latest supported nightly version by thise version of metals
+     */
+    private def findLatestSnapshot(exactVersion: String): Option[String] = try {
+
+      val metalsVersion = BuildInfo.metalsVersion
+
+      // strip timestamp to get only 3.2.2-RC1
+      val rcVersion = SemVer.Version
+        .fromString(exactVersion)
+        .copy(nightlyDate = None)
+        .toString()
+
+      val url =
+        s"https://oss.sonatype.org/content/repositories/snapshots/org/scalameta/"
```

Comment:
Would the same code work if we use `https://repo1.maven.org/maven2/org/scalameta/` as URL in case if it's a non-snapshot metals versions?

Sometimes we don't publish some nightlies because of action issues or it requires changes in cross-tests.

Reply:
For maven central we should be able to use coursier.complete, however I wouldn't want to revert to mtags published under latest stable version, since the interfaces of PresentationCompiler could have changed.